### PR TITLE
[ECH1976] Expose consumer member ID

### PR DESCRIPTION
**Pull request by alexcjing**
cc @KJTsanaktsidis @zendesk/echidna 

Paired with @Strainy 
Introduce MemberID method to Consumer so that we can attach this info to our log field.

https://zendesk.atlassian.net/browse/ECH-1976

### CODE REVIEW

File: kafka/consumer.go [hunk 1]
```diff
@@ -1062,3 +1062,10 @@ func (c *Consumer) handleRebalanceEvent(rkev *C.rd_kafka_event_t) (retval Event)
 
 	return nil
 }
+
+// MemberID returns this client's broker-assigned group member id.
+func(c *Consumer) MemberID() string {
```
Comment (alexcjing) [inline kafka/consumer.go:1067]: Is it worth testing this method? What is the best way to test this?

Comment (Strainy) [reply kafka/consumer.go:1067]: There's not a huge amount to test here. The only thing that might make sense to verify is that `rd_kafka_mem_free` is being called, but even still, I don't think there's a great deal of value in checking that.

File: kafka/consumer.go [hunk 2]
```diff
@@ -1062,3 +1062,10 @@ func (c *Consumer) handleRebalanceEvent(rkev *C.rd_kafka_event_t) (retval Event)
...
+	cMemberId := C.rd_kafka_memberid(c.handle.rk)
+	defer C.rd_kafka_mem_free(c.handle.rk, unsafe.Pointer(cMemberId))
```
Comment (KJTsanaktsidis) [inline kafka/consumer.go:1069]: I think this needs a null check before freeing - documentation says 

```
 * @returns An allocated string containing the current broker-assigned group
 *          member id, or NULL if not available.
 *          The application must free the string with \p free() or
 *          rd_kafka_mem_free()
```

And bad things tend to happen if you call free(NULL) ;)

File: kafka/consumer.go (hunk 1 above)
Comment (KJTsanaktsidis) [reply kafka/consumer.go:1067]: I could live without tests for this. It's hardly worth the difficulty of standing up the test suite...

**Comment by KJTsanaktsidis** [comment]
Other than the null check, looks good!

Can you open a PR upstream with this change as well? It should be pretty straightforward for them to merge I would have thought.

File: kafka/consumer.go (hunk 2 above)
Comment (alexcjing) [reply kafka/consumer.go:1069]: Thanks for the review @KJTsanaktsidis 🙇‍♀️ 
Based on[ Is it good practice to free a NULL pointer in C?](https://stackoverflow.com/questions/6084218/is-it-good-practice-to-free-a-null-pointer-in-c) which quoted the C standard, 7.20.3.2/2 from[ ISO-IEC 9899](http://www.open-std.org/JTC1/SC22/wg14/www/docs/n1124.pdf) 

> If ptr is a null pointer, no action occurs.

It seems that it's okay to not check NULL, do you think it applies to our case?

Comment (KJTsanaktsidis) [reply kafka/consumer.go:1069]: Huh, TIL! Looks like rd_kafka_mem_free just calls free()

https://github.com/edenhill/librdkafka/blob/6a1595f5e28a3627e2e0efc1bf28ea3dcdb094c7/src/rdkafka.c#L4883
https://github.com/edenhill/librdkafka/blob/063a9ae7a65cebdf1cc128da9815c05f91a2a996/src/rd.h#L138

so this should be OK.

**Comment by alexcjing** [comment]
Upstream PR https://github.com/confluentinc/confluent-kafka-go/pull/699

**Review by Strainy — APPROVED** [review]
Nice work navigating this one Alex!

**Review by KJTsanaktsidis — APPROVED** [review]

**Pull request merged**